ast/cinaps/gen_viewer.ml

```diff
@@ -38,6 +54,44 @@ module Structure : VIEWER_PRINTER = struct
       let vars = List.mapi tyl ~f:(fun i _ -> Printf.sprintf "arg%d" i) in
       Ml.tuple vars
 
+  (** Returns the list of types until a non-wrapper type is reached.
+      It's used to determine which `to_concrete` functions should be applied
+      until the result can be matched over.
+      E.g. if you consider a grammar describing:
+      [{
+      type ('a, 'b) t = {a : 'a; b = 'b}
+      type 'a u = ('a, int) t
+      type v = int u
+      }]
+      [unwrap_chain ~wrapper_types ~name:"v" ~ty:(Instance ("u", [Int])]
+      returns [("v", ...); ("u", ...)] because a value of type [t] can be
+      ppx_view-matched with the [a'match] and [b'match] views. *)
```

Comment:
It is indeed `a'match` and `b'match` since those are the names of the view functions over `t` values fields.

If you have a suggestion to make it clearer I'm happy to change it, I agree it's not great.

I wanted to emphasize that you should stop converting to concrete whenever the value can be matched over and that it meant different things for record and variant types since the view functions for those expect the abstract value.

Reply:
Oh, I see what it's saying now. I thought it was saying it returns `["v"; "u"]` and explaining why `v` and `u` are in the list. Instead it's explaining why `t` is not in the list. Let me propose an alternate version of the last two lines above:

```
returns ["v"; "u"] because values of type `v` can only be matched on by unrolling aliases `v` and `u`.
It stops without including `t` because `t` is a record type. It can be matched based on the fields `a` and `b`.
```

Reply:
That's much better indeed! Just updated it!